haskell/experimental/private/module.bzl

```diff
@@ -133,6 +138,26 @@ def _build_haskell_module(
     src = moduleAttr.src.files.to_list()[0]
     extra_srcs = [f for t in moduleAttr.extra_srcs + ctx.attr.extra_srcs for f in t.files.to_list()]
 
+    user_ghcopts = []
+    user_ghcopts += haskell_library_expand_make_variables("ghcopts", ctx, ctx.attr.ghcopts)
+
+    module_extra_attrs = [
+        [moduleAttr.src],
+        moduleAttr.extra_srcs,
+        moduleAttr.plugins,
+        moduleAttr.tools,
+    ]
+
+    user_ghcopts += expand_make_variables("ghcopts", ctx, moduleAttr.ghcopts, module_extra_attrs)
+
+    if src.extension == "hsc":
+        # TODO[AH] Support version macros
+        hsc_flags, hsc_inputs = preprocess_hsc_flags_and_inputs(dep_info, user_ghcopts, None)
+
+        # TODO[GL] Use idir when providing HaskellReplLoadInfo
```

Comment:
Not sure what to do here. I'd need to construct a repl provider to return from here, but we aren't currently doing so (and I'm not sure we can construct one from a single module?)

Reply:
The REPL providers are generated in the REPL aspect: https://github.com/tweag/rules_haskell/blob/f2a0c08425e49150063e9d65310d6766c1b0953e/haskell/repl.bzl#L547
We'll need to extend that aspect to support module targets.

Reply:
Sounds like something for another issue to me. I didn't find one open though, but maybe I'm just not recognising that some issue directly relates to this.

Reply:
Yes, definitely worth a ticket. I think we don't have a dedicated one, yet. There is a list of similar topics [here](https://github.com/tweag/rules_haskell/issues/1552#issuecomment-1000147400). But, I agree that they should be dedicated tickets.

Reply:
Created #1732